src/js/components/tabs.js

```diff
@@ -0,0 +1,186 @@
+*/
+import dispatchCustomEvent from '../utilities/dispatchCustomEvent';
+import { isNode, nodeListToArray } from '../utilities/domHelpers';
+import keyCodes from '../utilities/keyCodes';
+
+export default class Tabs {
+  constructor(element) {
+
+    // Instance properties
+    this.element = element;
+    this.tablist = this.element.querySelector('[role="tablist"]');
+    this.tabAttribute = 'data-tab';
+    this.tabSelector = `[${this.tabAttribute}]`;
+    this.tabs = nodeListToArray(this.element.querySelectorAll(this.tabSelector));
+    this.panelAttribute = 'data-tab-panel';
+    this.panelSelector = `[${this.panelAttribute}]`;
+    this.panels = nodeListToArray(this.element.querySelectorAll(this.panelSelector));
+
+
+    // Determine if a specific panel has been initialized with the data-tab-init attribute, otherwise, use the first tab
+    let initialPanel;
+    this.panels.forEach((panel, index) => {
+      if (panel.hasAttribute('data-tab-init')) {
+        initialPanel = panel;
+      } else {
+        this.tabs[index].setAttribute('tabindex', '-1');
+      }
+    })
+
+    // If a specific panel was initialized set this.openOnInit equal to it, otherwise fallback to the first panel
+    this.openOnInit = initialPanel || this.panels[0];
+
+    // bind methods
+    this._handleClick = this._handleClick.bind(this);
+    this._handleKeydown = this._handleKeydown.bind(this);
+
+    // Check to make sure that a DOM element was passed in for initialization
+    if (!isNode(this.element)) {
+      throw new TypeError(
+        'A DOM element should be passed as the first argument to initialize the modal.'
+      );
+    }
+
+    this.init();
+  }
+
+  /**
+   * This function is used to handle all click events on
+   * the document. It accepts the Event object, checks target to see if it came
+   * from a tab. From there, it gets the related panel and activates it.
+   * @param {Event} event
+   */
+  _handleClick(event) {
+    const currentTab = event.target.closest(this.tabSelector);
+    // If not a tab, ignore
+    if (!currentTab) return;
+
+    // Get the data-tab value
+    const currentTabValue = currentTab.getAttribute(this.tabAttribute);
+
+    // Get the associated panel
+    const currentPanel = this.element.querySelector(`[${this.panelAttribute}="${currentTabValue}"]`);
+
+    // Activate tab
+    this.activateTab(currentPanel);
+
+  }
+
+  /**
+   * This function is used to handle all keydown events on the document. It
+   * sets up handling for arrows, end, and home.
+   * @param {Event} event
+   */
+  _handleKeydown(event) {
+    const currentTab = event.target.closest(this.tabSelector);
+    // If not a tab, ignore
+    if (!currentTab) return;
+
+    // Create an array of all the focusable elements within the modal
+    const nextTab = this.tabs.indexOf(currentTab) + 1;
+    const prevTab = this.tabs.indexOf(currentTab) - 1;
+
+    switch (event.keyCode) {
+      case keyCodes.right || keyCodes.down: {
+
```

Comment:
Since there is only one statement (plus a `break`) in this case statements let's go ahead and close up the empty lines, and instead and line breaks between each `case {}` block. Right now there are no empty spaces between.

So as an example these first two would become:

```js
case keyCodes.right || keyCodes.down: {
  !this.tabs[nextTab] ? this.tabs[0].focus() : this.tabs[nextTab].focus();
  break;
}

case keyCodes.down: {
  !this.tabs[nextTab] ? this.tabs[0].focus() : this.tabs[nextTab].focus();
  break;
}
```